### PR TITLE
DOC: Add "pip install -r test_requirements.txt"

### PR DESCRIPTION
**Pull request by bhavukkalra**
Added as a Pre-Step for testing builds.

This Pull-Request is created in response to Addition - 2 suggested in issue #21156 

Before  this Addition the Page Looked Like - 

before

After the Addition the Page Looks like - 

upload

### CODE REVIEW

File: doc/source/user/building.rst [hunk 1]
```diff
@@ -94,10 +94,23 @@ Testing
 -------
 
 Make sure to test your builds. To ensure everything stays in shape, see if
-all tests pass::
+all tests pass
+
+First install all the python dependencies that will be required for testing 
+by running the below command in the root directory::
```
Comment (rossbar) [inline doc/source/user/building.rst:100]:
```suggestion
all tests pass.

First install all the Python dependencies that will be required for testing 
by running the below command in the root directory::
```

IMO this recommendation could be simplified further with something like:

```rst
The test suite requires additional dependencies which can easily be installed with::

    python -m pip install -r test_requirements.txt
```

The grammatical fixes are necessary, but feel free to ignore the wording suggestion.

File: doc/source/user/building.rst [hunk 2]
```diff
@@ -94,10 +94,23 @@ Testing
...
+Run tests::
 
     $ python runtests.py -v -m full
 
+``-v`` flag for showing the output of tests
+
+``-m`` flag for running a module or a package as a script
+
+``full`` for complete testing (will also run tests that are marked as slow)
+
```
Comment (rossbar) [inline doc/source/user/building.rst:113]: It seems the flags have gotten confused here. The `-m` flag is for `runtests.py`, not `python`, and is short for the `--mode` flag, of which `full` is one of the options.

**Review by rossbar — CHANGES_REQUESTED** [review]

File: doc/source/user/building.rst (hunk 1 above)
Comment (bhavukkalra) [reply doc/source/user/building.rst:100]: Got it. I am open to keeping the wording suggestion suggested here, I will surely check what comes in the next commit to be grammatically correct.


The only thing that I found on the official docs [Python Interpreter](https://docs.python.org/3/tutorial/interpreter.html) is that `-m` is used to run a module as a script. 

Just one query is `python -m pip install -r test_requirements.txt` a safer option or `pip install -r test_requirements.txt` is also correct?

File: doc/source/user/building.rst (hunk 2 above)
Comment (bhavukkalra) [reply doc/source/user/building.rst:113]: Correcting the flags as follows

`-m full` 
`-m` flag is for `runtests.py` shorthanded notation for `--mode` flag. 
`full` complete testing mode (will also run tests that are marked as slow) 

Could you please confirm?

Comment (Mukulikaa) [reply doc/source/user/building.rst:113]: Just a thought: I think these explanations belong in [testing builds](https://numpy.org/devdocs/dev/development_environment.html#testing-builds) to keep all information consolidated.

**Comment by bhavukkalra** [comment]
The latest commit with Title - 
DOC: Made suggested changes

Removed Additional flag info, Grammatical errors corrected.

Before - 

before

After suggested changes - 

final_upload

We can try to push those flags explanations to [testing builds](https://numpy.org/devdocs/dev/development_environment.html#testing-builds) page as suggested by @Mukulikaa. Do let me know if we want a separate PR for it or I should just make the changes in this single PR?

**Review by rossbar — APPROVED** [review]
LGTM, thanks @bhavukkalra! I agree with @Mukulikaa as well that paring this down was the right call.

**Pull request merged**